polaris-migrator/src/utilities/sass.ts

```diff
@@ -251,3 +263,398 @@ export function createInlineComment(text: string, options?: {prose?: boolean}) {
 
   return comment;
 }
+
+interface PluginOptions extends Options, NamespaceOptions {
+  partialFixStrategy: 'fix' | 'report' | undefined | false;
+  // Only applies when fixing
+  __injectReportsAsComments: boolean;
+}
+
+interface Report {
+  node: PostCSSNode;
+  severity: 'warning' | 'error' | 'suggestion';
+  message: string;
+}
+
+interface PluginContext {
+  fix: boolean;
+}
+
+// Extracted from stylelint
+type StylelintRuleBase<P = any, S = any> = (
+  primaryOption: P,
+  secondaryOptions: {[key: string]: S},
+  context: PluginContext,
+) => (root: Root) => void;
+
+interface StylelintRuleMeta {
+  url: string;
+  deprecated?: boolean;
+  fixable?: boolean;
+}
+
+type StylelintRule<P = any, S = any> = StylelintRuleBase<P, S> & {
+  ruleName: string;
+  meta?: StylelintRuleMeta;
+};
+
+export type PolarisMigrator<P = any, S = any> = (
+  primaryOption: P,
+  secondaryOptions: {
+    options: {[key: string]: S};
+    methods: {
+      report: (report: Report) => void;
+      each: <T extends Container>(
+        root: T,
+        walker: (decl: PostCSSNode) => false | void,
+      ) => void;
+      walk: <T extends Container>(
+        root: T,
+        walker: (decl: PostCSSNode) => false | void,
+      ) => void;
+      walkComments: <T extends Container>(
+        root: T,
+        walker: (comment: PostCSSComment) => false | void,
+      ) => void;
+      walkAtRules: <T extends Container>(
+        root: T,
+        atRuleWalker: (
+          atRule: AtRule,
+          parsedValue: ParsedValue,
+        ) => false | void,
+      ) => void;
+      walkDecls: <T extends Container>(
+        root: T,
+        declWalker: (
+          decl: Declaration,
+          parsedValue: ParsedValue,
+        ) => false | void,
+      ) => void;
+      walkRules: <T extends Container>(
+        root: T,
+        ruleWalker: (decl: PostCSSRule) => false | void,
+      ) => void;
+    };
+  },
+  context: PluginContext,
+) => (root: Root) => void;
+
+// Expose a stylelint-like API for creating sass migrators so we can easily
+// migrate to that tool in the future.
+function convertStylelintRuleToPostcssProcessor(ruleFn: StylelintRule) {
```

Comment:
This is the shim function.

Approx 90% of this function would be deleted by migrating to `stylelint`, meaning this function is just a shim to get `postcss`'s API to look like `stylelint`'s (for now).